File: Day_one_p1.py
with open("input.txt") as f:
    lines = f.readlines()

arr = []
for i in lines:
    s = ""
    for j in range(len(i)):
        if i[j] in "1234567890":
            s += i[j]
            break
    for j in range(len(i) - 1, -1, -1):
        if i[j] in "1234567890":
            s += i[j]
            break
    arr.append(int(s))
print((sum(arr)))

File: Day_one_p1.cpp
#include <bits/stdc++.h>
#include <fstream>

using namespace std;

int main()
{
    ifstream inputFile("1input.txt");
    vector<string> lines;
    string line;
    while (getline(inputFile, line))
    {
        lines.push_back(line);
    }
    inputFile.close();
    // for (auto i : lines)
    // {
    //     cout << i << " ";
    // }
    int fin = 0;
    for (int z = 0; z < lines.size(); z++)
    {
        string cur = lines[z];
        string ans = "";
        for (int i = 0; i < cur.size(); i++)
        {
            if (isdigit(cur[i]))
            {
                ans = ans + cur[i];
                break;
            }
        }
        for (int i = cur.size() - 1; i >= 0; i--)
        {
            if (isdigit(cur[i]))
            {
                ans = ans + cur[i];
                break;
            }
        }

        fin += stoi(ans);
    }

    cout << fin << endl;
    return 0;
}